window-copy.c

```diff
@@ -3083,10 +3101,19 @@ window_copy_search(struct window_mode_entry *wme, int direction, int regex,
 	wrapflag = options_get_number(wp->window->options, "wrap-search");
 	cis = window_copy_is_lowercase(str);
 
-	if (direction)
+	/*
+	 * Behave according to mode-keys. If it is emacs, search forward leaves the
+	 * cursor after the match. If it is vi, the cursor remains at the beginning of
+	 * the match, regardless of direction.
+	 */
+	keys = options_get_number(wp->window->options, "mode-keys");
+	if (direction) {
+		if (keys == MODEKEY_VI)
+			window_copy_move_right(s, &fx, &fy, wrapflag);
 		endline = gd->hsize + gd->sy - 1;
+	}
 	else {
-		if (again)
+		if (keys == MODEKEY_VI || again)
```

Comment:
Aside: the name of `again` is confusing to me and it doesn't appear to be set when I would expect. For example, AFAICT, an initial `C-s` sets `again` to 1, but it's not searching "again".